zeroconf/_dns.py

```diff
@@ -226,10 +226,18 @@ class DNSAddress(DNSRecord):
     """A DNS address record"""
 
     def __init__(
-        self, name: str, type_: int, class_: int, ttl: int, address: bytes, created: Optional[float] = None
+        self,
+        name: str,
+        type_: int,
+        class_: int,
+        ttl: int,
+        address: bytes,
+        created: Optional[float] = None,
+        scope_id: Optional[int] = None,
     ) -> None:
         super().__init__(name, type_, class_, ttl, created)
         self.address = address
+        self.scope_id = scope_id
```

Comment:
Does `__eq__` and `__hash__` need to be updated?

If the last record in the cache with the scope_id is the one that should always win then they don't since we will replace any existing records in the cache that match the address with the new scope id.

Since `ServiceInfo` gained the `interface_index` it seems like they don't need to change since the expectation is there can only be one.

Reply:
I think an interface index can change, if the interface is removed and then re-added. For example a USB ethernet adapter. You might only see this on Windows because interface indexes seems to be less deterministic there.

Anyway, implementing these dunder methods should be harmless, so I will.